Clinical-Genomics/scout
# Refactoring MatchMaker as a Scout extension

This PR adds a functionality:
- [x] Register MatchMaker integration as an extension when the app is created (fix #2375)
- [x] Remove many lines of code especially in the case views and controllers
- [x] Minimal MatchMaker views endpoints by moving all the processing in the controllers (fix #2149)
- [x] View MME button in the case side panel and make it visible (but not usable) to all users, so more people will be willing to use it
- [x] Write smarter tests


**How to test**:
1. Install on clinical-db stage 
1. Go to a case page and check that the case sidebar contains the matchmaker option:

![image](https://user-images.githubusercontent.com/28093618/115876258-5e330500-a446-11eb-982d-6c0449c09ca6.png)


- [ ] If your uses has no mme_submitter role, you shouldn't be allowed to interact with it
1. Use the command line to grant yourself permissions to use the matchmaker functionality:
`scout update user -u your_email -r mme_submitter`
1. Surf to a case that has been used previously for this purpose (tip: https://scout-stage.scilifelab.se/cust101/A1383)
- [ ] Check that you can submit new matches
- [ ] Check that you can review previous matches
- [ ] Check that you can remove this patient from MME
- [ ] Check that you can add/update this patient in MME

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
